src/Core/Runner.cpp

```diff
@@ -116,24 +116,28 @@ void Runner::runDetached(const QString &tmpFilePath, const QString &sourceFilePa
     setWorkingDirectory(tmpFilePath, sourceFilePath, lang);
 
     // different steps on different OSs
-#if defined(__unix__)
-    // use xterm on Linux
-    // check whether xterm is installed at first
-    LOG_INFO("Using xterm on unix");
+#if defined(Q_OS_LINUX)
+    QString terminal = SettingsHelper::getDetachedExecutionTerminal();
+
+    LOG_INFO("Using: " << terminal << " on Linux");
     QProcess testProcess;
-    testProcess.start("xterm", {"-v"});
+    testProcess.start(terminal, {"-v"});
     bool finished = testProcess.waitForFinished(2000);
     if (!finished || testProcess.exitCode() != 0)
     {
-        emit failedToStartRun(runnerIndex, tr("Please install xterm in order to use Detached Run."));
+        emit failedToStartRun(runnerIndex, QString(tr("Please install %1 in order to use detached run or change "
+                                                      "Detached Execution Terminal from Preferences."))
+                                               .arg(terminal));
         return;
     }
-    runProcess->setProgram("xterm");
-    runProcess->setArguments({"-e", getCommand(tmpFilePath, sourceFilePath, lang, runCommand, args) +
-                                        "; read -n 1 -s -r -p '\nExecution Done\nPress any key to exit'"});
-    LOG_INFO("Xterm args " << runProcess->arguments().join(" "));
+    runProcess->setProgram(terminal);
+    auto quotedCommand = getCommand(tmpFilePath, sourceFilePath, lang, runCommand, args);
+    QStringList execArgs = {
+        "-e", "/bin/bash -c '" + quotedCommand +
+                  " ; echo \"\nProgram completed with exit code $?\nPress any key to exit\" ; read -n 1'"};
```

Comment:
```suggestion
    QStringList execArgs = {"-e", QStringLiteral("/bin/bash -c '%1 ; echo \"\n%2\" ; read -n 1'")
                                      .arg(quotedCommand)
                                      .arg(tr("Program finished with exit code %1\nPress any key to exit").arg("$?"))};
```

`.arg("$?")` is used to prevent translators from translating `$?`.

Reply:
I'm not very sure whether UTF-8 is always supported in the terminal :thinking: It works fine on my Arch Linux with the Chinese translation of this.

Reply:
Most terminal emulators on Linux do support them. It also depends if fonts are installed. But I guess most people using Chinese or Russian as their language will have UTF-8 support in their emulator.